bindings/matrix-sdk-ffi/src/sliding_sync.rs

```diff
@@ -0,0 +1,580 @@
+    }
+}
+
+pub trait SlidingSyncObserver: Sync + Send {
+    fn did_receive_sync_update(&self, summary: UpdateSummary);
+}
+
+pub struct SlidingSync {
+    inner: matrix_sdk::SlidingSync,
+    client: Client,
+    observer: Arc<RwLock<Option<Box<dyn SlidingSyncObserver>>>>,
+    sync_handle: Arc<RwLock<Option<JoinHandle<()>>>>,
+}
+
+impl SlidingSync {
+    fn new(inner: matrix_sdk::SlidingSync, client: Client) -> Self {
+        SlidingSync { inner, client, observer: Default::default(), sync_handle: Default::default() }
+    }
+
+    pub fn set_observer(&self, observer: Option<Box<dyn SlidingSyncObserver>>) {
+        *self.observer.write().unwrap() = observer;
+    }
+
+    pub fn subscribe(
+        &self,
+        room_id: String,
+        settings: Option<RoomSubscription>,
+    ) -> anyhow::Result<()> {
+        let settings =
+            if let Some(settings) = settings { Some(settings.try_into()?) } else { None };
+        self.inner.subscribe(room_id.try_into()?, settings);
+        Ok(())
+    }
+
+    pub fn unsubscribe(&self, room_id: String) -> anyhow::Result<()> {
+        self.inner.unsubscribe(room_id.try_into()?);
+        Ok(())
+    }
+
+    pub fn get_view(&self, name: String) -> Option<Arc<SlidingSyncView>> {
+        for s in self.inner.views.lock_ref().iter() {
+            if s.name == name {
+                return Some(Arc::new(SlidingSyncView { inner: s.clone() }));
+            }
+        }
+        None
+    }
+
+    pub fn get_room(&self, room_id: String) -> anyhow::Result<Option<Arc<SlidingSyncRoom>>> {
+        Ok(self
+            .inner
+            .get_room(OwnedRoomId::try_from(room_id)?)
+            .map(|inner| Arc::new(SlidingSyncRoom { inner, client: self.client.clone() })))
+    }
+
+    pub fn get_rooms(
+        &self,
+        room_ids: Vec<String>,
+    ) -> anyhow::Result<Vec<Option<Arc<SlidingSyncRoom>>>> {
+        let actual_ids = room_ids
+            .into_iter()
+            .map(OwnedRoomId::try_from)
+            .collect::<Result<Vec<OwnedRoomId>, IdParseError>>()?;
+        Ok(self
+            .inner
+            .get_rooms(actual_ids.into_iter())
+            .into_iter()
+            .map(|o| {
+                o.map(|inner| Arc::new(SlidingSyncRoom { inner, client: self.client.clone() }))
+            })
+            .collect())
+    }
+
+    pub fn sync(&self) -> Arc<StoppableSpawn> {
+        let inner = self.inner.clone();
+        let observer = self.observer.clone();
+        let spawn = Arc::new(StoppableSpawn::with_handle_ref(self.sync_handle.clone()));
+        let inner_spawn = spawn.clone();
+        {
+            let mut sync_handle = self.sync_handle.write().unwrap();
+
+            if let Some(handle) = sync_handle.take() {
+                handle.abort();
+            }
+
+            *sync_handle = Some(RUNTIME.spawn(async move {
+                let stream = inner.stream().await.expect("Doesn't fail.");
```

Comment:
The “expect message” is… vague. 2 options: updating the message, or just use `unwrap `.

```suggestion
                let stream = inner.stream().await.unwrap();
```